pctiler/pctiler/reader.py

```diff
@@ -44,36 +44,41 @@ def __init__(self, request: Request):
 @attr.s
 class ItemSTACReader(PgSTACReader):
 
-    # TODO: remove CustomCOGReader once moved to rasterio 1.3
-    reader: Type[BaseReader] = attr.ib(default=CustomCOGReader)
-
     # We make request an optional attribute to avoid re-writing
     # the whole list of attribute
     request: Optional[Request] = attr.ib(default=None)
 
-    def _get_asset_url(self, asset: str) -> str:
-        asset_url = BlobCDN.transform_if_available(super()._get_asset_url(asset))
+    def _get_asset_info(self, asset: str) -> AssetInfo:
+        """return asset's url."""
+        asset_url = BlobCDN.transform_if_available(
+            super()._get_asset_info(asset)["url"]
+        )
 
         if self.input.collection_id:
             render_config = get_render_config(self.input.collection_id)
             if render_config and render_config.requires_token:
                 asset_url = pc.sign(asset_url)
 
-        return asset_url
+        asset_info = self.input.assets[asset]
+        info = AssetInfo(url=asset_url)
+
+        if "file:header_size" in asset_info.extra_fields:
+            h = asset_info.extra_fields["file:header_size"]
+            info["env"] = {"GDAL_INGESTED_BYTES_AT_OPEN": h}
+
+        return info
```

Comment:
we now return a TypedDict with the URL and a dictionary used to set some GDAL env (if available)